src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2395,5 +2395,36 @@ private bool notifyMethodInfoUsage(CORINFO_METHOD_STRUCT_* ftn)
         {
             return true;
         }
+
+        private CORINFO_CLASS_STRUCT_* getStaticFieldCurrentClass(CORINFO_FIELD_STRUCT_* field, byte* pIsSpeculative)
+        {
+            Debug.Assert(pIsSpeculative == null); // To be deleted
+
+            FieldDesc fieldDesc = HandleToObject(field);
+
+            if (fieldDesc.IsStatic && !fieldDesc.IsThreadStatic && fieldDesc.OwningType is MetadataType owningType &&
+                !owningType.IsCanonicalSubtype(CanonicalFormKind.Any))
+            {
+                // Generally, this API is only used for reference types to provide better information
+                // to the JIT for possible devirtualizations. Value types are typically handled
+                // separately and folded to constants via getStaticFieldContent. However, some
+                // fields aren't foldable (e.g. ExternSymbolMappedField), so let's tell the JIT
+                // it can rely on them being invariant too.
+                if (fieldDesc.HasRva)
+                {
+                    // Read-only RVA fields need no "is class initialized" check.
```

Comment:
How do you know that the RVA field is read-only?

Reply:
> How do you know that the RVA field is read-only?

Can it be not so? I thought RVA is mutable only for C++/CLI, etc. Here we check that the field is final.

Reply:
We have several of these compiler generated RVA fields. Some of them are read-only, some of them are mutable.

> Here we check that the field is final.

Where do we check that?